pkg/alertmanager/operator.go

```diff
@@ -737,18 +738,39 @@ func (c *Operator) UpdateStatus(ctx context.Context, key string) error {
 		return fmt.Errorf("failed to retrieve statefulset state: %w", err)
 	}
 
+	selectorLabels := makeSelectorLabels(a.Name)
+	selector, err := metav1.LabelSelectorAsSelector(&metav1.LabelSelector{MatchLabels: selectorLabels})
+	if err != nil {
+		return fmt.Errorf("failed to create selector for alertmanager scale status: %w", err)
+	}
+
+	a.Status.Selector = selector.String()
+	a.Status.Replicas = ptr.Deref(a.Spec.Replicas, 1)
```

Comment:
the field is already set by stsReporter.Update()

Reply:
Thanks, I wondered what it does? I mean does it update the status subresource selector field based on replicas?

Reply:
among other things yes.